php/Exchange.php

```diff
@@ -2163,24 +1950,33 @@ public function market_ids ($symbols) {
         return array_map (array ($this, 'market_id'), $symbols);
     }
 
-    public function marketIds ($symbols) {
-        return $this->market_ids ($symbols);
-    }
-
     public function market_id ($symbol) {
         return (is_array ($market = $this->market ($symbol))) ? $market['id'] : $symbol;
     }
 
-    public function marketId ($symbol) {
-        return $this->market_id ($symbol);
+    public function __call ($function, $params) {
+        if (array_key_exists ($function, $this->defined_rest_api)) {
+            $api_call_info = $this->defined_rest_api[$function];
+            $entry = $api_call_info[3];
+            $api_call_info[3] = $params;
```

Comment:
Shouldn't it be `$api_call_info = array ($params);` here?

Reply:
With variable params passing them to base methods won't be too easy. But I will keep the `__call` for `$this->defined_rest_api`.